cranelift-interpreter/src/frame.rs

```diff
@@ -0,0 +1,123 @@
+//! Implements a call frame (activation record) for the Cranelift interpreter.
+
+use crate::value::Value;
+use cranelift_codegen::ir::{Function, Value as ValueRef};
+use std::collections::HashMap;
+
+/// Holds the mutable elements of an interpretation. At some point I thought about using
+/// Cell/RefCell to do field-level mutability, thinking that otherwise I would have to
+/// pass around a mutable object (for inst and registers) and an immutable one (for function,
+/// could be self)--in the end I decided to do exactly that but perhaps one day that will become
+/// untenable.
+#[derive(Debug)]
+pub struct Frame<'a> {
+    /// The currently executing function.
+    pub function: &'a Function,
```

Comment:
Is this better as a `FuncRef`? I used `&'a Function` here because it was convenient but it may be more consistent and (possibly?) efficient to use a 4-byte `FuncRef` here instead. (I probably need to look again at how efficient it is to use `FuncRef` to index into an `Environment`).